src/com/mie/controller/AppServlet.java

```diff
@@ -56,50 +68,147 @@ protected void doGet(HttpServletRequest request,
 		 */
 		String forward = "";
 		String action = request.getParameter("action");
+		System.out.println(action);
 		HttpSession session = request.getSession();
-		int userId = Integer.parseInt((String) session.getAttribute("userId"));
+//		int userId = Integer.parseInt((String) session.getAttribute("userId"));
+		int userId = (Integer) session.getAttribute("userId");
+//		try {
+//			userId = (Integer) request.getAttribute("userId");
+//		} catch (Exception e) {
+//			userId = (Integer) session.getAttribute("userId");
+//		}
+		
+		
+		if (action!=null) {
+			if (action.equalsIgnoreCase("DeleteApp")) {
+				int appId = Integer.parseInt(request.getParameter("appId"));
+				appdao.deleteApplication(appId);
+				
+			} else if (action.equalsIgnoreCase("InsertApp")) {
+				System.out.println("Inserting App for user" + userId);
+				Application app = new Application();
+				//app.setAppId(Integer.parseInt(request.getParameter("appId")));
+				app.setCompany((request.getParameter("company")));
+				app.setUserId(userId);
+				app.setUrl((request.getParameter("url")));
+				String date = request.getParameter("date");
+				Date date1 = Date.valueOf(date);
+				app.setDeadline(date1);
+				try {
+					app.setInterview(Integer.parseInt(request.getParameter("interview")));
+				} catch (Exception e) {
+					app.setInterview(0);
+				}
+				try {
+					app.setJob(Integer.parseInt(request.getParameter("joboffer")));
+				} catch (Exception e) {
+					app.setJob(0);
+				}
+				app.setIndustry(request.getParameter("industry"));
+				app.setPosition(request.getParameter("position"));
+				
+				appdao.addApplication(app);
+				
+			} else if (action.equalsIgnoreCase("UpdateApp")) {
+				Application app = new Application();
+				app.setAppId(Integer.parseInt(request.getParameter("appId")));
+				app.setCompany((request.getParameter("company")));
+				app.setUserId(userId);
+				app.setUrl((request.getParameter("url")));
+				String date = request.getParameter("date");
```

Comment:
just heads up that the date value has to be inputted in yyyy-[m]m-[d]d format ([https://www.javatpoint.com/java-sql-date-valueof-method](https://www.javatpoint.com/java-sql-date-valueof-method))